### PR TITLE
algol68g 3.0.4

### PR DESCRIPTION
**Pull request by Porkepix**
Created with `brew bump-formula-pr`.

### CODE REVIEW

**Comment by cho-m** [comment]
Clang failed during configure.

GCC-11 gets past that, but now failing during build:
```
  gcc-11 -DHAVE_CONFIG_H -I.  -I./src/include  -DBINDIR='"/opt/homebrew/Cellar/algol68g/3.0.4/bin"' -DINCLUDEDIR='"/usr/local/include/algol68g"' -g -O2 -Wall -Wshadow -Wunused-variable -Wunused-parameter -Wno-long-long -fno-diagnostics-color -MT ./src/a68g/a68g-char.o -MD -MP -MF ./src/a68g/.deps/a68g-char.Tpo -c -o ./src/a68g/a68g-char.o `test -f './src/a68g/char.c' || echo './'`./src/a68g/char.c
  In file included from /opt/homebrew/Cellar/gcc/11.2.0_3/lib/gcc/11/gcc/aarch64-apple-darwin20/11/include-fixed/stdio.h:421,
                   from ./src/include/a68g-includes.h:32,
                   from ./src/include/a68g.h:31,
                   from ./src/a68g/a68glib.c:27:
  ./src/a68g/a68glib.c: In function 'a68_snprintf':
  ./src/a68g/a68glib.c:46:7: error: expected declaration specifiers or '...' before numeric constant
     46 |   int vsnprintf (char *, size_t, const char *, va_list);
        |       ^~~~~~~~~
  In file included from /Library/Developer/CommandLineTools/SDKs/MacOSX11.sdk/usr/include/secure/_stdio.h:31,
                   from /opt/homebrew/Cellar/gcc/11.2.0_3/lib/gcc/11/gcc/aarch64-apple-darwin20/11/include-fixed/stdio.h:421,
                   from ./src/include/a68g-includes.h:32,
                   from ./src/include/a68g.h:31,
                   from ./src/a68g/a68glib.c:27:
  ./src/a68g/a68glib.c:46:7: error: expected declaration specifiers or '...' before '__builtin_object_size'
     46 |   int vsnprintf (char *, size_t, const char *, va_list);
        |       ^~~~~~~~~
```

**Comment by cho-m** [comment]
Added fixes/workarounds to build Linux and Intel macOS.

A previous CI run failed on ARM macOS due to `isfinite/finite` support not detected:
```
  checking for isfinite... no
  checking for finite... no

...

gcc-11 -DHAVE_CONFIG_H -D_FORTIFY_SOURCE=0 -I.  -I./src/include  -DBINDIR='"/opt/homebrew/Cellar/algol68g/3.0.4/bin"' -DINCLUDEDIR='"/opt/homebrew/Cellar/algol68g/3.0.4/include/algol68g"' -g -O2 -Wall -Wshadow -Wunused-variable -Wunused-parameter -Wno-long-long -fno-diagnostics-color -c -o ./src/a68g/a68g-socket.o `test -f './src/a68g/socket.c' || echo './'`./src/a68g/socket.c
  ./src/a68g/single-math.c:113:2: error: #error "cannot define a68_finite"
    113 | #error "cannot define a68_finite"
        |  ^~~~~
  make: *** [src/a68g/a68g-single-math.o] Error 1
  make: *** Waiting for unfinished jobs....
```

Intel outputs something like:
```
  checking for isfinite... no
  checking for finite... yes
```

**Comment by SMillerDev** [comment]
This should really be upstreamed at least as issues before we merge this.

File: Formula/algol68g.rb [hunk 1]
```diff
@@ -28,12 +28,54 @@ class Algol68g < Formula
...
+
+    inreplace "Makefile.in" do |s|
+      # Work around the hardcoded include directory
+      s.gsub! "= /usr/local/include/algol68g", "= #{include}/algol68g"
+
+      # Work around macOS build error. Env variables like LDFLAGS, CFLAGS, DEFS aren't picked up.
+      # ./src/a68g/a68glib.c:46:7: error: expected declaration specifiers or '...' before numeric constant
+      #    46 |   int vsnprintf (char *, size_t, const char *, va_list);
+      #       |       ^~~~~~~~~
+      s.gsub!(/^DEFS =.*$/, "\\0 -D_FORTIFY_SOURCE=0") if OS.mac?
```
[the remark this answers is not in the document]
Comment (cho-m) [reply Formula/algol68g.rb:70]: Passing to `configure` or environment variables doesn't work.

Passing to `make` may override the existing values. I think `DEFS` and `CFLAGS` are both assigned to something already.

[the remark this answers is not in the document]
Comment (carlocab) [reply Formula/algol68g.rb:70]: Yea, fair enough. We could probably simplify this `inreplace` with something like `s.get_make_var("DEFS")` and `s.change_make_var!` though.

**Comment by chenrui333** [comment]
#123264

**Pull request closed without merging**